### PR TITLE
AGENT-889: Integration tests for add-nodes command

### PR DESCRIPTION
**Pull request by andfasano**
(This patch is based on PR #8093, so the relevant commit is [1e6ab8](https://github.com/openshift/installer/commit/1e6ab8a93db4364ec97d2abf4e0fd2956ae79e6a))

This patch introduces an integration test for the `node-joiner add-nodes` command. Similarly to the already existing openshift-installer integration tests, it adopts [testscript](https://pkg.go.dev/github.com/rogpeppe/go-internal/testscript). In addition, since the command requires the presence of a cluster, it uses [envTest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest) for that scope.
To properly arrange the test scenario required, all the yaml file generated by the test in the `setup` subfolder will be pre-processed and applied to the cluster before running the test.

### CODE REVIEW

**Comment by andfasano** [comment]
Note: in order to pass, the `agent-integration-tests` job needs to use an image where capi binaries are present

File: cmd/node-joiner/nodejoiner_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	}
+
+	projectDir, err := os.Getwd()
+	assert.NoError(t, err)
+
+	// Prepare the temporary cluster that will be used by node-joiner.
+	testEnv := &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			// Preload OpenShift specific CRDs so that it will be possible to create specific OpenShift resources.
+			filepath.Join(build.Default.GOPATH, "pkg", "mod", "github.com", "openshift", "api@v0.0.0-20240301093301-ce10821dc999", "config", "v1"),
```
Comment (rwsu) [inline cmd/node-joiner/nodejoiner_integration_test.go:58]: Was seeing:
```
[rwsu@installer]$ go test ./cmd/node-joiner/.
--- FAIL: TestNodeJoinerIntegration (0.07s)
    --- FAIL: TestNodeJoinerIntegration/add-nodes (5.58s)
        testscript.go:534: FAIL: testdata/add-nodes.txt:0: unable to install CRDs onto control plane: unable to read CRD files: stat /home/rwsu/go/pkg/mod/github.com/openshift/api@v0.0.0-20240301093301-ce10821dc999/config/v1: no such file or directory
```
If you having troubling running the new integration test, I had to do the following:
```
export OPENSHIFT_INSTALL_CLUSTER_API=YES
./hack/build.sh
go mod vendor
go mod tidy
go test ./cmd/node-joiner/.
```

File: cmd/node-joiner/testdata/add-nodes.txt [hunk 2]
```diff
@@ -0,0 +1,134 @@
+# Verify that the add-nodes command generates correctly the ISO image
+
+exec node-joiner add-nodes --kubeconfig=$WORK/kubeconfig --log-level=debug --dir=$WORK
+exists $WORK/agent-addnodes.x86_64.iso
+
```
Comment (rwsu) [inline cmd/node-joiner/testdata/add-nodes.txt:5]: Should we also assert the contents of agent-addnodes.x86_64.iso contain the correct values in /etc/assisted/manifests to add the extra-worker-0 node to the cluster?

**Review by rwsu — CHANGES_REQUESTED** [review]

**Comment by andfasano** [comment]
Replaced by https://github.com/openshift/installer/pull/8976

**Pull request closed without merging**